### ISSUE TITLE
Add ranks for plant hybrids to taxonRank vocabulary

### ISSUE THREAD

**Issue by qgroom** [gbif/rs.gbif.org#79]
The vocabulary for taxonRank lacks ranks for plant hybrids.
i.e. nothofamily, nothosubfamily, nothotribe, nothosubtribe, nothogenus, nothosubgenus, nothospecies, nothosubspecies

However, these are valid rank terms for plant hybrids as stated in the International Code of Nomenclature for algae, fungi, and plants article [H.3.1](https://www.iapt-taxon.org/nomen/pages/main/art_h3.html)

**Comment by mdoering** [comment]
Indeed. And I agree we should allow them to be shared as such. 

Both the GBIF & COL API separate those values into 2 values, the rank without notho and a separate notho field which represents the part of the name which is prefixed by the x. See also https://github.com/tdwg/dwc/issues/43 and a COL API example: http://api.catalogueoflife.org/dataset/3LR/taxon/94NJ

With that model both APIs rank enumeration should not contain the notho values above.

**Comment by qgroom** [comment]
@mdoering So the notho field is in effect a hybrid flag, at least for plants, fungi and algae.
What is the advantage of this rather than treating these as additional rank terms?

I should have mentioned that this DwC issue is also raised. https://github.com/tdwg/dwc/issues/390

**Comment by mdoering** [comment]
One advantage is that we can compare ranks more easily. One goal of our rank enumeration is to provide a native ordering. With "duplicates" like species and nothospecies this becomes tricky.

Also the notho field tells us where to place the hybrid marker. If I remember the code correctly you either specify a notho rank marker OR use the hybrid marker, but never both. They are equivalent. 

`Foo bar nothovar. baz` = `Foo bar var. × baz` if I am not mistaken.

**Comment by amueller-bgbm** [comment]
I agree with @mdoering that adding notho ranks has disadvantages for ordering. We also implement hybrids with according flags (for each name part) and correct formatting if one of the flags is set. So the usage of "notho" and "×" is more about correct formatting the hybrid information and not about changing the semantics of the rank used.
